dhall/src/Dhall.hs

```diff
@@ -1341,11 +1374,45 @@ instance (Selector s1, Selector s2, Interpret a1, Interpret a2) => GenericInterp
 
         return (Type {..})
 
-instance Interpret a => GenericInterpret (M1 S s (K1 i a)) where
-    genericAutoWith options = do
-        let Type { extract = extract', ..} = autoWith options
+instance (Selector s, Interpret a) => GenericInterpret (M1 S s (K1 i a)) where
+    genericAutoWith options@InterpretOptions{..} = do
+        let n :: M1 S s (K1 i a) r
+            n = undefined
+
+        name <- fmap fieldModifier (getSelName n)
```

Comment:
I'm wondering whether applying the `fieldModifier` first is a good idea. This means that the `Smart` option depends it. It should be documented at least.